source/org/zfin/gwt/curation/server/FeatureRPCServiceImpl.java

```diff
@@ -393,11 +405,12 @@ public FeatureDTO editFeatureDTO(FeatureDTO featureDTO) throws DuplicateEntryExc
     private boolean featureLocationNeedsUpdate(FeatureDTO featureDTO, FeatureLocation fgl) {
         //null safe
         return !(
-                    StringUtils.equals(featureDTO.getFeatureChromosome(), fgl.getChromosome()) &&
-                    StringUtils.equals(featureDTO.getFeatureAssembly(), fgl.getAssembly()) &&
-                    Objects.equals(featureDTO.getFeatureStartLoc(), fgl.getStartLocation()) &&
-                    Objects.equals(featureDTO.getFeatureEndLoc(), fgl.getEndLocation())
-                );
+            StringUtils.equals(featureDTO.getFeatureChromosome(), fgl.getChromosome()) &&
+                StringUtils.equals(featureDTO.getFeatureAssembly(), fgl.getAssembly()) &&
+                Objects.equals(featureDTO.getFeatureStartLoc(), fgl.getStartLocation()) &&
+                Objects.equals(featureDTO.getFeatureEndLoc(), fgl.getEndLocation()) &&
+                Objects.equals(featureDTO.getEvidence(), FeatureService.getFeatureGenomeLocationEvidenceCode(fgl.getLocationEvidence().getZdbID()))
```

Comment:
Is there a chance of NullPointerException for fgl.getLocationEvidence().getZdbID() ?

Reply:
Not in our case, the calling method is checked but it would be good to handle a null evidence code if someone called this method.